### PR TITLE
Index class primary constructor parameters

### LINKED ISSUE #134: Mtags does not index primary constructor fields
screen shot 2017-12-17 at 13 49 01
screen shot 2017-12-17 at 13 49 08

We can handle these cases here https://github.com/scalameta/language-server/blob/1ebc5392f2a29de09825fa34347d60a0aad69142/metaserver/src/main/scala/scala/meta/languageserver/mtags/ScalaMtags.scala#L30

### PR DESCRIPTION
**Pull request by olafurpg**
Fixes #134. I noticed that the new "clear index" command doesn't cleanup the in-memory `indexedJars` map forcing me to restart the server to re-index.

This is my first PR developed (almost) entirely through vscode 🎉

### CODE REVIEW

**Comment by gabro** [comment]
>  I noticed that the new "clear index" command doesn't cleanup the in-memory indexedJars map forcing me to restart the server to re-index.

Good point, I remind thinking of that and I eventually decided that clearing and restarting was good enough for that PR. We may want to add a command on-top-of that to also trigger re-indexing.

File: metaserver/src/main/scala/scala/meta/languageserver/mtags/ScalaMtags.scala [hunk 1]
```diff
@@ -27,7 +27,13 @@ object ScalaMtags {
             term(t.name, PACKAGEOBJECT);
             term("package", t.name.pos, OBJECT);
             continue()
-          case t: Defn.Class => tpe(t.name, CLASS); continue()
+          case t: Defn.Class =>
+            tpe(t.name, CLASS)
+            for {
+              params <- t.ctor.paramss
+              param <- params
+            } withOwner() { super.param(param.name, VAL) }
```
Comment (laughedelic) [inline metaserver/src/main/scala/scala/meta/languageserver/mtags/ScalaMtags.scala]: I noticed that there is `PARAM` flag, would it fit better here or is it for something else?

Comment (olafurpg) [reply metaserver/src/main/scala/scala/meta/languageserver/mtags/ScalaMtags.scala]: Done! I'm not sure how it we can use it in the language server, but adding one more bit of information is free :)

Comment (olafurpg) [reply metaserver/src/main/scala/scala/meta/languageserver/mtags/ScalaMtags.scala]: I'm actually being lazy here by adding `VAL` blindly to everything even for non-val parameter. I added a note!

**Pull request merged**